design/design_template.md

```diff
@@ -0,0 +1,101 @@
+---
```

Comment:
Part of me doesn't want to introduce a template, but as long as we keep it simple I see that it will be useful. Templates that are too long are just too tedious and not really helpful. Can we simplify this to the bare minimum? Even just titles for the sections, and remove descriptions that should be obvious.